fe/fe-core/src/main/java/org/apache/doris/nereids/rules/implementation/LogicalOlapScanToPhysicalOlapScan.java

```diff
@@ -31,12 +44,35 @@ public class LogicalOlapScanToPhysicalOlapScan extends OneImplementationRuleFact
     @Override
     public Rule build() {
         return logicalOlapScan().then(olapScan ->
-                // TODO: olapScan should get (OlapTable);
-                new PhysicalOlapScan(
-                    (OlapTable) olapScan.getTable(),
-                    olapScan.getQualifier(),
-                    Optional.empty(),
-                    olapScan.getLogicalProperties())
+            new PhysicalOlapScan(
+                (OlapTable) olapScan.getTable(),
```

Comment:
add a Override function in LogicalOlapScan, and then remove cast on `olapScan.getTable()`
```java
@Override
OlapTable getTable();
```